fiftyone/brain/internal/core/redaction.py

```
@@ -0,0 +1,430 @@
+
+| Copyright 2017-2025, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+"""
+import os
+import logging
+import shutil
+from typing import Tuple
+
+import numpy as np
+import cv2
+
+from eta.core.video import VideoProcessor
+
+import fiftyone as fo
+import fiftyone.core.storage as fos
+import fiftyone.core.brain as fob
+import fiftyone.core.labels as fol
+import fiftyone.core.validation as fov
+from fiftyone import ViewField as F
+
+
+logger = logging.getLogger(__name__)
+
+
+_ALLOWED_TYPES = (
+    fol.Detection,
+    fol.Detections,
+)
+
+_GAUSSIAN_BLUR_KERNEL_SIZE = lambda w: (w // 10) * 2 + 1  # ensure ksize is odd
+_STACK_BLUR_KERNEL_SIZE = lambda w: min(w, 50) * 2 + 1  # ensure ksize is odd
+
+
+def create_redaction(
+    samples,
+    label_field,
+    label_classes,
+    redaction_type,
+    redaction_method,
+    redaction_field,
+    force_recreate,
+    progress,
+):
+    """See ``fiftyone/brain/__init__.py``."""
+
+    #
+    # Algorithm
+    #
+    # Generates a blurred/masked image based on the redaction_method.
+    # Identifies areas of the image (based on the redaction_type = bounding_box/segmentation_mask)
+    # corresponding to the label_classes from the label_field
+    # and replaces the regions with the redaction_method image
+    #
+
+    fov.validate_collection(samples)
+    fov.validate_collection_label_fields(samples, label_field, _ALLOWED_TYPES)
+
+    config = RedactionConfig(
+        label_field,
+        label_classes,
+        redaction_type,
+        redaction_method,
+        redaction_field,
+        force_recreate,
+    )
+    brain_key = config.redaction_field
+    brain_method = config.build()
+    brain_method.ensure_requirements()
+    brain_method.register_run(samples, brain_key, cleanup=False)
+    brain_method.register_samples(samples)
+
+    select_fields = [
+        label_field,
+        brain_key,
+    ]
+    select_fields = [
+        ff for ff in select_fields if samples.get_field(ff) is not None
```

I'm confused what redaction_field is. Is it valid for it not to exist? Should you throw an error instead of just filtering it out?

---

1. thanks for bringing my attention here, this should be `f"{config.redaction_field}_filepath"` not config.redaction_path (a.k.a. `brain_key`)
2. `f"{config.redaction_field}_filepath"` is the target field to which we are saving the new redacted images.
3. If this exists, we may edit it; if it doesn't, we create it.
I added the field creation earlier, so that I can use `set_values`